contracts/upgradeable_contracts/native_to_erc20/ForeignBridgeNativeToErc.sol

```diff
@@ -0,0 +1,64 @@
+pragma solidity 0.4.24;
+import "../../libraries/SafeMath.sol";
+import "../BasicBridge.sol";
+import "../../IBurnableMintableERC677Token.sol";
+import "../../ERC677Receiver.sol";
+import "../BasicForeignBridge.sol";
+import "openzeppelin-solidity/contracts/token/ERC20/ERC20Basic.sol";
+
+
+contract ForeignBridgeNativeToErc is ERC677Receiver, BasicBridge, BasicForeignBridge {
+    /// Event created on money withdraw.
+    event UserRequestForAffirmation(address recipient, uint256 value);
+
+    function initialize(
+        address _validatorContract,
+        address _erc677token,
+        uint256 _dailyLimit,
+        uint256 _maxPerTx,
+        uint256 _minPerTx,
+        uint256 _foreignGasPrice,
+        uint256 _requiredBlockConfirmations
+    ) public returns(bool) {
+        require(!isInitialized());
+        require(_validatorContract != address(0));
+        require(_minPerTx > 0 && _maxPerTx > _minPerTx && _dailyLimit > _maxPerTx);
+        require(_foreignGasPrice > 0);
+        addressStorage[keccak256(abi.encodePacked("validatorContract"))] = _validatorContract;
+        setErc677token(_erc677token);
+        uintStorage[keccak256(abi.encodePacked("dailyLimit"))] = _dailyLimit;
+        uintStorage[keccak256(abi.encodePacked("deployedAtBlock"))] = block.number;
+        uintStorage[keccak256(abi.encodePacked("maxPerTx"))] = _maxPerTx;
+        uintStorage[keccak256(abi.encodePacked("minPerTx"))] = _minPerTx;
+        uintStorage[keccak256(abi.encodePacked("gasPrice"))] = _foreignGasPrice;
+        uintStorage[keccak256(abi.encodePacked("requiredBlockConfirmations"))] = _requiredBlockConfirmations;
+        setInitialize(true);
+        return isInitialized();
+    }
+
+    function onTokenTransfer(address _from, uint256 _value, bytes /*_data*/) external returns(bool) {
+        require(msg.sender == address(erc677token()));
+        require(withinLimit(_value));
+        setTotalSpentPerDay(getCurrentDay(), totalSpentPerDay(getCurrentDay()).add(_value));
+        erc677token().burn(_value);
+        emit UserRequestForAffirmation(_from, _value);
+        return true;
+    }
+
+    function claimTokensFromErc677(address _token, address _to) external onlyOwner {
+        erc677token().claimTokens(_token, _to);
+    }
+
+    function erc677token() public view returns(IBurnableMintableERC677Token) {
+        return IBurnableMintableERC677Token(addressStorage[keccak256(abi.encodePacked("erc677token"))]);
+    }
+
+    function onExecuteMessage(address _recipient, uint256 _amount) internal returns(bool){
+        return erc677token().mint(_recipient, _amount);
+    }
+
+    function setErc677token(address _token) private {
```

Comment:
Is my understanding correct and we can move this functionality in a separate contract like `ERC677Management` and inherit it on `ForeignBridge` contract in case of `coin2token` bridge and on `HomeBridge` contract in case of `token2token` bridge?